### ISSUE TITLE
bug: `targets.genericLinux` breaks Plasma on Arch Linux

### ISSUE THREAD

**Issue by metiulekm** [nix-community/home-manager#4682]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

After a recent update of Arch and my HM config, I found that (X11) Plasma would not start. Some Plasma systemd services would start, but the initial splash screen would never end, except fading into black in some cases. This happened on all my computers. After some debugging, I found that removing `targets.genericLinux.enable = true;` from my config, switching the config and stopping any remaining background programs before attempting to login in again would fix the issue.

I understand that `targets.genericLinux` provides config that is appropriate for non-NixOS installations. It's not crucial to me, so for now I have just commented it out in my config.

I believe that the issue happened after Arch update and not HM/nixpkgs update, since I have also reproduced the issue with the ~100 days old versions of HM and nixpkgs.

Here's a minimal `flake.nix` that I have been able to reproduce the issue on a fresh user account with:
```nix
{
  description = "Home Manager configuration";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
    home-manager = {
      url = "github:nix-community/home-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };

  outputs = inputs: let
    system = "x86_64-linux";
    pkgs = import inputs.nixpkgs {
      inherit system;
      allowAliases = false;
    };
  in {
    homeConfigurations.metiulekm = inputs.home-manager.lib.homeManagerConfiguration {
      inherit pkgs;
      modules = [
        (
          {
            config,
            ...
          }: {
            home = {
              username = "test";
              homeDirectory = "/home/${config.home.username}";
              stateVersion = "22.05";
            };

            targets.genericLinux.enable = true;
          }
        )
      ];
    };

    devShell.${system} = pkgs.mkShell {
      buildInputs = [inputs.home-manager.packages.${system}.home-manager];
    };
  };
}
```

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 6.6.1-zen1-1-zen, Arch Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"nixpkgs"`
 - channels(metiulekm): `""`
 - nixpkgs: `/nix/store/mrckm9b3v6wg2fljdp3z4pja1sz07g3z-source`
```

**Comment by metiulekm** [comment]
> Did you get any solution to boot into your system again?

Yeah, the other TTYs (Ctrl+Alt+F\*) worked just fine. I were able to login to a shell session there, remove the line and `home-manager switch …`.

**Comment by hexclover** [comment]
Hello, just to mention that I ran into (almost) the same issue when doing a fresh install of `home-manager` on my Arch box, except even the "splash screen" wouldn't show up, just a black screen.

In my case `targets.genericLinux.enable` option was irrelevant, and what I had to do was to downgrade `shared-mime-info` from 2.4 (which arrived on Arch on 2023-11-12) to 2.3 (the same as in nixpkgs) instead. Seems the co-existence of these two versions in `XDG_DATA_DIRS` is confusing KSycoca.

**Comment by metiulekm** [comment]
> For me setting `xdg.mime.enable = false;` in home-manager also works (`targets.genericLinux.enable` can be left as `true`). Can be used instead of downgrading host system's `shared-mime-info`.

Finally found some mana to check this out and this is also true for me, great find @hexclover :) I guess that `targets.genericLinux.enable` was the only thing in my config that required `shared-mime-info` or something, not sure. I guess I will rewrite the original description (though what you've found might mean that this should probably be fixed in Qt anyway? I have no idea…).

**Comment by hexclover** [comment]
> I guess I will rewrite the original description (though what you've found might mean that this should probably be fixed in Qt anyway? I have no idea…).

Me neither. I do think it's a Qt bug though, and I have no idea what home-manager or nixpkgs can do to fix it. Maybe we should just wait for https://github.com/NixOS/nixpkgs/pull/267850 to eventually make it into master and unstable, and hopefully even 23.11. In the meanwhile perhaps just keep this issue open so that people who are affected can come across a fix here.

I may file a bug to Qt when I have time, although I'm not sure if they will fix it, as it may seem strange to have two `shared-mime-info` in `XDG_DATA_DIRS`, and even such a situation does not necessarily introduce cycles.
